sdk/ai/azure-ai-inference/azure/ai/inference/prompts/core.py

```diff
@@ -0,0 +1,333 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""
+Adapted from https://github.com/langchain-ai/langchain
+MIT License
+"""
+from __future__ import annotations
+
+import abc
+import json
+import os
+import re
+from pathlib import Path
+from typing import Any, Dict, Generic, List, Literal, Optional, Type, TypeVar, Union
+
+import yaml
+from pydantic import BaseModel, ConfigDict, Field, FilePath
```

Comment:
`pydantic` is (still) a problematic dependency due to its versioning/major versions. Note to self; check if there are any versioning constraints (e.g. >=pydantic v2) that can cause conflicts with other libraries.